xdsl/dialects/scf.py

```diff
@@ -148,9 +155,13 @@ def get(
 class For(IRDLOperation):
     name = "scf.for"
 
-    lb: Operand = operand_def(IndexType)
-    ub: Operand = operand_def(IndexType)
-    step: Operand = operand_def(IndexType)
+    T = Annotated[
+        Attribute, AnyOf([IndexType, SignlessIntegerConstraint]), ConstraintVar("T")
+    ]
```

Comment:
That's where you could write `Annotated[IndexType | SignlessIntegerType, ConstraintVar("T")]`

Reply:
I tried this and other variations but I get:

```py
TypeError: unhashable type: 'ParamAttrConstraint'
```

from the `typing` module.

Reply:
Alternatively, I'd move this in `builtin` and name it `AnySignlessIntegerOrIndexType` like in MLIR along with naming the above to `AnySignlessIntegerType` instead of `SignlessIntegerType`.

WDYT @math-fehr

Reply:
@math-fehr I've done it the way I wrote above.